controllers/avatar.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * CRUD avatar image (middlewares)
+ */
+
+const sharp = require('sharp'),
+      fs = require('fs-extra'),
+      path = require('path'),
+      crypto = require('crypto'),
+      Identicon = require('identicon.js'),
+      _ = require('lodash');
+
+const multer = require('multer');
+const upload = multer({ dest: 'uploads/' });
+
+const models = require(path.resolve('./models'));
+
+/**
+ * GET user's avatar.
+ * Will return either image uploaded by user (cropped and formatted to jpeg),
+ * or default svg
+ */
+async function get(req, res, next) {
+  const { username } = req.params;
+
+  const usernameExists = await models.user.exists(username);
```

Comment:
True, the database can be off, etc. We need to handle this.

Reply:
The potential errors need to be passed to an error handling middleware via `next(error)`;
Otherwise the request will timeout, probably. And in the future the application would fail (as uncaught Promise rejections are deprecated).